### PR TITLE
flattening data tree + renaming

### LINKED ISSUE #1071: Flatten data + coredoc trees
More info: https://hackmd.io/BXsOO8GTQpGomkweh-8Yvg?both

(work on feature branch)

### PR DESCRIPTION
**Pull request by mikiquantum**
Closes #1071

### CODE REVIEW

File: documents/coredocument.go [hunk 1]
```diff
@@ -437,44 +428,29 @@ func (cd *CoreDocument) DocumentRootTree(docType string, dataRoot []byte) (tree
 	return tree, nil
 }
 
-// signingRootTree returns the merkle tree for the signing root.
-func (cd *CoreDocument) signingRootTree(docType string, dataRoot []byte) (tree *proofs.DocumentTree, err error) {
-	if len(dataRoot) != idSize {
-		return nil, errors.NewTypedError(ErrCDTree, errors.New("data root is invalid"))
+// dataTree returns the merkle tree for the document data tree provided.
+func (cd *CoreDocument) dataTree(docType string, docDataTree *proofs.DocumentTree) (tree *proofs.DocumentTree, err error) {
```
Comment (lucasvo) [inline documents/coredocument.go]:
```suggestion
func (cd *CoreDocument) dataTree(docType string, embeddedLeaves []*proofs.LeafNode) (tree *proofs.DocumentTree, err error) {
```

File: documents/coredocument.go [hunk 2]
```diff
@@ -437,44 +428,29 @@ func (cd *CoreDocument) DocumentRootTree(docType string, dataRoot []byte) (tree
...
+	if docDataTree == nil {
+		return nil, errors.NewTypedError(ErrCDTree, errors.New("data tree is invalid"))
 	}
-
 	cdTree, err := cd.coredocTree(docType)
```
Comment (lucasvo) [inline documents/coredocument.go]:
```suggestion
	cdLeaves, err := proofs.FlattenMessage(....)
```

File: documents/coredocument.go [hunk 3]
```diff
@@ -437,44 +428,29 @@ func (cd *CoreDocument) DocumentRootTree(docType string, dataRoot []byte) (tree
...
-			Hashed:   true,
-		},
-		{
-			Property: NewLeafProperty(fmt.Sprintf("%s.%s", SigningTreePrefix, CDRootField), append(CompactProperties(SigningTreePrefix), CompactProperties(CDRootField)...)),
-			Hash:     cdTree.RootHash(),
-			Hashed:   true,
-		},
-	})
 
+	err = tree.AddLeaves(append(docDataTree.GetLeaves(), cdTree.GetLeaves()...))
```
Comment (lucasvo) [inline documents/coredocument.go]:
```suggestion
	err = tree.AddLeaves(append(cdLeaves, embeddedLeaves...))
```

**Comment by vedhavyas** [comment]
@mikiquantum can you point me to the spec for this change in HackMD, I dont seem to find it.

File: documents/entityrelationship/model.go [hunk 4]
```diff
@@ -240,29 +240,29 @@ func (e *EntityRelationship) AddNFT(grantReadAccess bool, registry common.Addres
 
 // CalculateSigningRoot calculates the signing root of the document.
 func (e *EntityRelationship) CalculateSigningRoot() ([]byte, error) {
-	dr, err := e.CalculateDataRoot()
+	t, err := e.getDocumentDataTree()
 	if err != nil {
-		return dr, err
+		return nil, errors.New("failed to get data tree: %v", err)
```
Comment (vedhavyas) [inline documents/entityrelationship/model.go]: This seems to repeat. Add a typed error for this?

**Comment by mikiquantum** [comment]
@vedhavyas https://hackmd.io/BXsOO8GTQpGomkweh-8Yvg?both#Replace-current-signing-root-with-merged-data-and-coredoc-tree

**Pull request merged**